src/poetry/console/commands/add.py

```diff
@@ -169,6 +169,9 @@ def handle(self) -> int:
                 assert isinstance(version, str)
                 parse_constraint(version)
 
+            if "subdirectory" in _constraint and _constraint["subdirectory"] is None:
+                del _constraint["subdirectory"]
```

Comment:
I’m not quite sure what this `del` does (and others like it), but perhaps 77d897586c5f1f036bc27e2cfb75b1517062b0c0 has made it unnecessary?